api/src/main/java/org/apache/brooklyn/api/typereg/ManagedBundle.java

```diff
@@ -27,9 +27,13 @@
 
     /** A URL-like thing that we can register with the OSGi framework
      * to uniquely identify this bundle-instance.
-     * This typically includes the unique {@link #getId()} of this item. */
+     * This typically includes the unique {@link #getId()} of this item. 
+     * This will not normally be a URL that can be loaded. */
     String getOsgiUniqueUrl();
 
     VersionedName getVersionedName();
 
+    /** MD5 checksum of the contents of bundle as installed to OSGi */
+    String getChecksum();
```

Comment:
it uses `md5` but it has some opinions about what it uses for checksum so better to consider it its own checksum